app/main.py

```diff
@@ -1,20 +1,39 @@
-from typing import Optional
-from fastapi import Body, FastAPI
-from pydantic import BaseModel
+from fastapi import Depends, FastAPI, status, HTTPException
+from sqlalchemy.orm import Session
+from sqlalchemy import text
+
+from app.models import User
+from .database.commands import db_create_user, db_get_users, db_get_user
+from .database import db_models
+from .database.setup import engine, get_db
 
 fastapi = FastAPI()
 
-class User(BaseModel):
-    username: str
-    password: str
-    email: Optional[str] = None
-    phone_number: Optional[int] = None
+get_db()
+db_models.Base.metadata.create_all(bind=engine)
+
 
 @fastapi.get("/")
 def root():
     return {"message: Welcome to my API"}
 
-@fastapi.post("/users")
-def create_user(new_user: User):
-    print(new_user)
-    return { "message": "New User created", "user_info": new_user.dict() }
\ No newline at end of file
+@fastapi.post("/users", status_code=status.HTTP_201_CREATED)
+def create_user(new_user: User, db: Session = Depends(get_db)):
+    user = db_create_user(new_user, db)
+    return { "message": "New User created", "user_info": user}
+
+@fastapi.get("/users")
+def get_users(db: Session = Depends(get_db)):
+    users = db_get_users(db)
+    if not users:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,
+                            detail = f"No users found")
+    return {"users": users}
+
+@fastapi.get("/users/{username}")
+def get_user_by_username(username: str, db: Session = Depends(get_db)):
+    user = db_get_user(username, db)
+    if not user:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,
+                            detail = f"user '{username}' not found")
```

Comment:
I would have `db_get_user` raise the error instead of doing it here then everywhere you use the function you don't need to keep repeating this snippet.